lemmy.yml

```diff
@@ -21,15 +21,54 @@
     - setup: # gather facts
 
   tasks:
+  - name: Install aptitude
+    apt:
+      name: aptitude
+      state: latest
+      update_cache: true
+
   - name: install dependencies
     apt:
-      update_cache: yes
+      state: latest
+      update_cache: true
       pkg:
         - 'nginx'
-        - 'docker-compose'
-        - 'docker.io'
         - 'certbot'
         - 'python3-certbot-nginx'
+        - 'apt-transport-https'
+        - 'ca-certificates'
+        - 'curl'
+        - 'software-properties-common'
+        - 'python3-pip'
+        - 'virtualenv'
+        - 'python3-setuptools'
+
+  - name: Add Docker GPG apt Key
+    apt_key:
+      url: https://download.docker.com/linux/ubuntu/gpg
+      state: present
+
+  - name: Add Docker Repository
+    apt_repository:
+      repo: deb https://download.docker.com/linux/ubuntu focal stable
+      state: present
+
+  - name: Update apt and install docker-ce
+    apt:
+      name: docker-ce
+      state: latest
+      update_cache: true
+
+  - name: Install Docker Module for Python
+    pip:
+      name: docker
+
+  - name: Install or upgrade docker-compose
+    get_url: 
+      url : "https://github.com/docker/compose/releases/download/1.25.5/docker-compose-Linux-x86_64"
+      dest: /usr/local/bin/docker-compose
+      mode: 'a+x'
+      force: yes
```

Comment:
Why are you switching the Docker install method? Afaik there is no problem with Docker from Ubuntu repos and it has the major advantage that its very easy to install and upgrade.

Reply:
I tried on a brand new test server, and it completely fails now. Using the docker repos on ubuntu or debian is never the recommended way anymore, and it asks you to install using their official method. This was taken from an ansible example that uses that official install method.

Reply:
Hmm alright.